### ISSUE TITLE
ReflectionTypeLoadException handle null message

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#15710]
New failure in
`baseservices/exceptions/regressions/V1/SEH/coverage/Exceptions/Exceptions.sh`

```
Throwing AmbiguousMatchException..
Caught the exception: Humpty Dumpty sat on a wall..
Throwing ReflectionTypeLoadException..

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Reflection.ReflectionTypeLoadException.get_Message()
   at MightyExceptor.Main(String[] Args)
```

@dotnet/arm64-contrib FYI

**Comment by sdmaclea** [comment]
```
git bisect bad 55f9cc3a
git bisect good 0950d37d8
```
@wtgodbe Looks like it was introduced by #15688

**Comment by wtgodbe** [comment]
I suspect it is because https://github.com/dotnet/coreclr/blob/2597bacae161cb81590863f887343daec837bb81/src/mscorlib/shared/System/Reflection/ReflectionTypeLoadException.cs#L54 tries to iterate through a 1-length Exception array containing nothing -
 https://github.com/dotnet/coreclr/blob/master/tests/src/baseservices/exceptions/regressions/V1/SEH/coverage/Exceptions.cs#L297. So https://github.com/dotnet/coreclr/blob/2597bacae161cb81590863f887343daec837bb81/src/mscorlib/shared/System/Reflection/ReflectionTypeLoadException.cs#L56 would be trying to dereference null. @jkotas  @danmosemsft AggregateException does the same thing here - https://github.com/dotnet/coreclr/blob/eb2e9f90ba3679fa8386c4130f81c8061e8d4ca4/src/mscorlib/shared/System/AggregateException.cs#L437. Is this a bad test, or a case where we need to fix the implementation of the Exceptions?

**Comment by jkotas** [comment]
This is a bug in your change.

AggregateException constructor ensures that the individual exceptions are not null. 

ReflectionTypeLoadException constructor does not ensure that the individual exceptions are not null, and changing it to throw for these would be a breaking change.

I think you should fix this by skipping null entries while trying to format the message.

**Comment by BruceForstall** [comment]
Because this is causing test failures, you should consider reverting it now and re-PR'ing later with the fix.

**Comment by wtgodbe** [comment]
I just put up the PR https://github.com/dotnet/coreclr/pull/15711, would it be preferable for me to Revert my old change, or is it alright to just merge that PR once CI is green?

**Comment by wtgodbe** [comment]
https://github.com/dotnet/coreclr/pull/15708 is also up which modifies the same file

**Comment by jkotas** [comment]
Merge #15708 first, and then resolve conflict and merge #15711 once the CI is green. This is straightforward enough fix - no need to revert.

**Comment by danmosemsft** [comment]
What flavor of CI job would have caught this failure?

**Comment by BruceForstall** [comment]
I'm not sure if it only exists on arm64 or every platform; presumably every.

Perhaps it only happens in the pri-1 test runs?

It seems we currently have problems where we're not consistently or correctly running pri-1 where we think we are; hopefully https://github.com/dotnet/coreclr/pull/15075 will fix that.

**Comment by jkotas** [comment]
This was fixed by #15711. @wtgodbe Could you please add regression tests to corefx before closing the issue?

**Comment by wtgodbe** [comment]
https://github.com/dotnet/corefx/pull/26155 added tests
